app/controllers/locations_controller.rb

```diff
@@ -48,6 +48,19 @@ def destroy
     end
   end
 
+  def weeks_exclude_dates
+    start_date = params[:start].to_date
+    end_date = params[:end].to_date - 5.days
```

Comment:
Tests are failing because `params[:start]` is nil. Perhaps include a guard clause: 

```
return if dates_params[:start].blank? || dates_params[:end].blank?
...

def dates_params
   params.require(:location).permit(:start, :end, :exclude_dates)
end